onnxruntime/core/session/custom_ops.cc

```diff
@@ -562,7 +562,8 @@ Status IsCompatible(const ONNX_NAMESPACE::OpSchema& schema, const OrtCustomOp* o
     const auto characteristic = op->GetInputCharacteristic(op, i);
     const auto& formal_parameter = input_parameters[i];
     if (characteristic == OrtCustomOpInputOutputCharacteristic::INPUT_OUTPUT_OPTIONAL) {
-      ORT_RETURN_IF_NOT(formal_parameter.GetOption() == onnx::OpSchema::FormalParameterOption::Optional,
+      ORT_RETURN_IF_NOT(op->version < min_ort_version_with_optional_io_support ||
```

Comment:
should we separate the condition into two, other the error message is a little misleading?

Reply:
the condition will always be met if the version is lower than supported, meaning the check is getting skipped.
otherwise, we do the check, and report error is not met. This is neater than a "if () {}", isn't it?